contributors/design-proposals/local-storage-overview.md

```diff
@@ -0,0 +1,600 @@
+
+    ```yaml
+    apiVersion: v1
+    kind: pod
+    metadata:
+     name: foo
+    spec:
+     containers:
+     - name: fooc
+       resources:
+         limits:
+           storage.kubernetes.io/logs: 200Mi
+           storage.kubernetes.io/overlay: 200Mi
+       volumeMounts:
+       - name: myEmptyDir
+         mountPath: /mnt/data
+     volumes:
+     - name: myEmptyDir
+       emptyDir:
+	     size: 1Gi
+    ```
+
+4. Bob’s “foo” pod can use upto “200Mi” for its containers logs and writable layer each, and “1Gi” for its “myEmptyDir” volume. 
+5. If Bob’s pod “foo” exceeds the “default” storage limits and gets evicted, then Bob can set a minimum storage requirement for his containers and a higher “capacity” for his EmptyDir volumes.
+
+  ```yaml
+  apiVersion: v1
+  kind: pod
+  metadata:
+   name: foo
+  spec:
+   containers:
+   - name: fooc
+     resources:
+       requests:
+         storage.kubernetes.io/logs: 500Mi
+         storage.kubernetes.io/overlay: 500Mi
+     volumeMounts:
+     - name: myEmptyDir
+       mountPath: /mnt/data
+   volumes:
+   - name: myEmptyDir
+     emptyDir:
+		size: 2Gi
+  ```
+
+6. It is recommended to require `limits` to be specified for `storage` in all pods. `storage` will not affect the `QoS` Class of a pod since no SLA is intended to be provided for storage capacity isolation. it is recommended to use Persistent Durable Volumes as much as possible and avoid primary partitions.
+
+### Alice manages a Database which needs access to “durable” and fast scratch space
+
+1. Cluster administrator provisions machines with local SSDs and brings up the cluster
+2. When a new node instance starts up, an addon DaemonSet discovers local “secondary” partitions which are mounted at a well known location and creates Local PVs for them if one doesn’t exist already. The PVs will include a path to the secondary device mount points, and a hostname label ties the volume to a specific node.  A StorageClass is required and will have a new optional field `toplogyKey` for the system to apply node constraints to local storage when scheduling pods that request this StorageClass.  Other labels may also be specified.
+
+    ```yaml
+    kind: StorageClass
+    apiVersion: storage.k8s.io/v1
+    metadata:
+      name: local-fast
+    provisioner: ""
+    toplogyKey: kubernetes.io/hostname
+    ```
+    ```yaml
+    kind: PersistentVolume
+    apiVersion: v1
+    metadata:
+      name: local-pv-1
+      labels:
+        kubernetes.io/hostname: node-1
+    spec:
+      capacity:
+        storage: 100Gi
+      local:
+        path: /var/lib/kubelet/storage-partitions/local-pv-1
+      accessModes:
+        - ReadWriteOnce
+      persistentVolumeReclaimPolicy: Delete
+      storageClassName: local-fast
+    ```
+    ```
+    $ kubectl get pv
+    NAME       CAPACITY ACCESSMODES RECLAIMPOLICY STATUS    CLAIM … NODE
+    local-pv-1 100Gi    RWO         Delete        Available         node-1
```

Comment:
The daemonset will operate like an external provisioner and handle the cleanup and reclaimation.  It will delete the contents of the PV, and then create a new PV for it.